pkg/controller/util/endpoint/controller_utils_test.go

```diff
@@ -191,8 +189,9 @@ func TestShouldPodBeInEndpointSlice(t *testing.T) {
 					PodIP: "1.2.3.4",
 				},
 			},
-			expected: true,
+			expected: false,
```

Comment:
this is a change in behaviour, but this was a bug, a failed pod with restart policy always is alwas in phase Running

Reply:
That is the expected behavior (restart always pods do not enter terminal phases).